packages/api/src/services/features.ts

```diff
@@ -0,0 +1,67 @@
+import { Feature } from '../entity/feature'
+import { getRepository } from '../entity/utils'
+import * as jwt from 'jsonwebtoken'
+import { env } from '../env'
+import { IsNull, Not } from 'typeorm'
+
+enum FeatureName {
+  UltraRealisticVoice = 'ultra-realistic-voice',
+}
+
+export const getFeatureName = (name: string): FeatureName | undefined => {
+  return Object.values(FeatureName).find((v) => v === name)
+}
+
+export const optInFeature = async (
+  name: FeatureName,
+  uid: string
+): Promise<Feature | undefined> => {
+  if (name === FeatureName.UltraRealisticVoice) {
+    return optInUltraRealisticVoice(uid)
+  }
+
+  return undefined
+}
+
+const optInUltraRealisticVoice = async (uid: string): Promise<Feature> => {
+  const feature = await getRepository(Feature).findOneBy({
+    user: { id: uid },
+    name: FeatureName.UltraRealisticVoice,
+  })
+  if (feature) {
+    // already opted in
+    console.log('already opted in')
+    return feature
+  }
+
+  // opt in to feature for the first 1000 users
+  const count = await getRepository(Feature).countBy({
+    name: FeatureName.UltraRealisticVoice,
+    grantedAt: Not(IsNull()),
+  })
+
+  let grantedAt: Date | null = new Date()
+  if (count >= 1000) {
+    console.log('feature limit reached')
+    grantedAt = null
+  }
+
+  return getRepository(Feature).save({
+    user: { id: uid },
+    name: FeatureName.UltraRealisticVoice,
+    grantedAt,
+  })
+}
+
+export const signFeatureToken = (feature: Feature): string => {
+  return jwt.sign(
+    {
+      userid: feature.user.id,
```

Comment:
Not sure if we need to copy but we use `uid` for a regular token.

Also, I'm not sure if the JWT library handles expiration itself if we use these names:

```
  const iat = Math.floor(Date.now() / 1000)
  const exp = Math.floor(
    new Date(Date.now() + 1000 * 60 * 60 * 24 * expireInDays).getTime() / 1000
  )
  return jwt.sign({ uid: userId, iat, exp }, env.server.jwtSecret)
```

Reply:
Yeah, the JWT library handles the `iat` and `exp` itself and I have updated the token to be expired in 1 day.
Do you think it is enough?

Reply:
Also renamed `userid` to `uid` in the JWT token